### PR TITLE
fix(streaming): avoid violating the assumption of U+ after U- in hash join & add checker

### LINKED ISSUE #1801: streaming: ensure `U+` `U-` appears in pairs
Currently, filter executor relies on the assumption that U+ U- appears in pairs. However, there are many places that this assumption is broken.

We should add a schema check that every single chunk from executor contains U+ U- in pairs. Some executors might need refactor to pass this check.

### PR DESCRIPTION
**Pull request by BugenZhao**
## What's changed and what's your intention?

When there're multiple rows matched with a single `UpdateDelete`, the hash join executor will yield multiple continuous `UpdateDelete`s, which violates the assumption of U+ after U- used by `Merge`, `Filter` and so on. One can simply reproduce this in tpch q17, q18 with rearranged chain in #1556 since it introduces more barriers than before.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
- Close #1801

### CODE REVIEW

File: src/stream/src/executor/debug/update_check.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+impl UpdateCheckExecutor {
+    pub fn new(input: Box<dyn Executor>) -> Self {
+        Self { input }
+    }
+}
+
+#[async_trait]
+impl super::DebugExecutor for UpdateCheckExecutor {
```
Comment (skyzh) [inline src/stream/src/executor/debug/update_check.rs:38]: Can we merge all debug executors into one? The backtrace would be disaster...

**Review by skyzh — APPROVED** [review]

**Comment by skyzh** [comment]
also test failed

**Review by MrCroxx — APPROVED** [review]
LGTM

File: src/stream/src/executor/debug/update_check.rs (hunk 1 above)
Comment (BugenZhao) [reply src/stream/src/executor/debug/update_check.rs:38]: Next time for sure!

**Review by yuhao-su — APPROVED** [review]

**Pull request merged**